trantor/net/inner/TimerQueue.cc

```diff
@@ -270,12 +270,27 @@ void TimerQueue::reset(const std::vector<TimerPtr> &expired,
     loop_->assertInLoopThread();
     for (auto const &timerPtr : expired)
     {
-        if (timerPtr->isRepeat() &&
-            timerIdSet_.find(timerPtr->id()) != timerIdSet_.end())
+        // if (timerPtr->isRepeat() &&
+        //     timerIdSet_.find(timerPtr->id()) != timerIdSet_.end())
+        // {
+        //     timerPtr->restart(now);
+        //     insert(timerPtr);
+        // }
+        if (timerIdSet_.find(timerPtr->id()) != timerIdSet_.end())
```

Comment:
Usually, I use an iterator for this, because it may be used later:
```c++
auto iter=timerIdSet_.find(timerPtr->id());
if(iter!=timerIdSet_.end())
{
      if().....
      else
      {
           timerIdSet_.erase(iter); //use an iterator to avoid a search same as before.
      }
}
```